File: src/main/com/stockcomp/repository/jpa/InvestmentOrderRepository.java
package com.stockcomp.repository.jpa;

import com.stockcomp.entity.contest.InvestmentOrder;
import com.stockcomp.entity.contest.OrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface InvestmentOrderRepository extends JpaRepository<InvestmentOrder, Long> {

    List<InvestmentOrder> findAllByOrderStatus(OrderStatus orderStatus);
}

File: src/main/com/stockcomp/investmentorder/InvestmentOrderRepository.kt
package com.stockcomp.investmentorder

import com.stockcomp.investmentorder.entity.InvestmentOrder
import com.stockcomp.investmentorder.entity.OrderStatus
import com.stockcomp.participant.entity.Participant
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface InvestmentOrderRepository : JpaRepository<InvestmentOrder, Long> {

    fun findAllByParticipantAndSymbolAndOrderStatus(
        participant: Participant, symbol: String, orderStatus: OrderStatus
    ): List<InvestmentOrder>

    fun findAllByParticipantAndOrderStatus(
        participant: Participant, orderStatus: OrderStatus
    ): List<InvestmentOrder>
}